lib/worker.js

```diff
@@ -54,10 +55,38 @@ async function processPackageFile(repoName, packageFile, packageConfig) {
   // Find all upgrades for remaining dependencies
   const upgrades = await findUpgrades(dependencies, config);
   // Process all upgrades sequentially
+  if (config.maintainYarnLock) {
+    await maintainYarnLock(config);
+  }
   await processUpgradesSequentially(config, upgrades);
   logger.info(`${repoName} ${packageFile} done`);
 }
 
+async function maintainYarnLock(inputConfig) {
+  const packageContent = await inputConfig.api.getFileContent(inputConfig.packageFile);
+  const yarnLockFileName = path.join(path.dirname(inputConfig.packageFile), 'yarn.lock');
+  logger.debug(`Checking for ${yarnLockFileName}`);
+  const existingYarnLock = await inputConfig.api.getFileContent(yarnLockFileName);
+  if (!existingYarnLock) {
+    return;
+  }
+  logger.debug('Found existing yarn.lock file');
+  const newYarnLock = await branchWorker.getYarnLockFile(packageContent, inputConfig);
+  if (existingYarnLock === newYarnLock.contents) {
+    logger.debug('Yarn lock file does not need updating');
+    return;
+  }
+  logger.debug('Yarn lock needs updating');
+  const commitMessage = 'Renovate yarn.lock file';
```

Comment:
Could this be added to the default definitions so that it can be configured by the project or `renovate` user? (We use Angular commit conventions for all our commits to help with other automation tooling)